### PR TITLE
upgrade erlang to version 26.2.3

### PR DESCRIPTION
**Pull request by sameluch**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Upgrade erlang to the latest version for azurelinux 3.0

###### Change Log  

- Upgrade erlang to version 26.2.3

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local Build
- [Buddy Build](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=532747&view=results)

### CODE REVIEW

File: SPECS/erlang/erlang.spec [hunk 1]
```diff
@@ -18,12 +18,12 @@ BuildRequires:  unzip
 erlang programming language
```
Comment (rikenm1) [inline SPECS/erlang/erlang.spec:18]: Just a nit. 
can we add this here? "Erlang is a programming language and runtime system for building massively scalable soft real-time systems with requirements on high availability."

**Review by rikenm1 — APPROVED** [review]
Just one nit. Looks good otherwise

**Review by anphel31 — APPROVED** [review]
Looks good, but any idea if we can enable the tests?
It looks like static tests are available:
`Make sure all static checks pass by calling ./otp_build check. `
https://github.com/erlang/otp/blob/master/CONTRIBUTING.md#before-you-submit-your-pull-request

**Comment by sameluch** [comment]
> Looks good, but any idea if we can enable the tests? It looks like static tests are available: `Make sure all static checks pass by calling ./otp_build check. ` https://github.com/erlang/otp/blob/master/CONTRIBUTING.md#before-you-submit-your-pull-request

I added the tests to the packages and have this buddy build running with the static tests enabled without docs https://dev.azure.com/mariner-org/mariner/_build/results?buildId=533469&view=results

File: SPECS/erlang/erlang.spec (hunk 1 above)
Comment (sameluch) [reply SPECS/erlang/erlang.spec]: Updated

**Pull request merged**